Improve RotatingKeysSSLEngineProviderSpec

jrudolph opened the pull request:
Attempt to fix #29221

jrudolph committed:
tests: increase timeout during termination

Attempt to fix #29221

raboof reviewed (approved):
Suspicious, but OK.

jrudolph committed:
tighten timeout and add debug logging

jrudolph commented:
PLS BUILD

ignasi35 commented on akka-remote/src/test/scala/akka/remote/artery/tcp/ssl/RotatingKeysSSLEngineProviderSpec.scala:
```diff
@@ -261,14 +264,17 @@ abstract class RotatingKeysSSLEngineProviderSpec(extraConfig: String)
+      system.log.info(s"Terminated $systemToTerminate after ${lasted / 1000000} ms")
     }
     // Don't cleanup folder until all systems have terminated
     cleanupTemporaryDirectory()
```
I like the idea of terminating the `systemsToTerminate` in `beforeTermination` instead of `afterTermination` but `cleanupTemporaryDirectory` must happen in `afterTermination` always: the system in testkit also uses the rotating keys implementation so it could poll the file system and fail when the folders have been deleted.

For the rest it LGTM.

ignasi35 commented:
We could also enable coordinated shutdown loge to try and pinpoint what's the phase that takes too long to shutdown.

ignasi35 commented:
> We could also enable coordinated shutdown logs to try and pinpoint what's the phase that takes too long to shutdown.

Oh, the latest change (enables `debug`) already prints the coordinated shutdown logs. I noticed the `actor-system-terminate` phase includes the transport shutdown:

```
[DEBUG] [06/15/2020 12:33:38.411] [systemB-reread-akka.actor.internal-dispatcher-6] [CoordinatedShutdown(akka://systemB-reread)] Performing task [terminate-system] in CoordinatedShutdown phase [actor-system-terminate]
[INFO] [06/15/2020 12:33:38.411] [systemB-reread-akka.remote.default-remote-dispatcher-16] [akka://systemB-reread@localhost:40569/system/remoting-terminator] Shutting down remote daemon.
[INFO] [06/15/2020 12:33:38.412] [systemB-reread-akka.remote.default-remote-dispatcher-16] [akka://systemB-reread@localhost:40569/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
```

and also that no other phase in the shutdown has any task. This looks like a timeout on the transport shutdown.

ignasi35 committed:
Stop using testkit's actorsystem. Cleaup after termination

ignasi35 commented:
@jrudolph I've changed the test code to avoid using the same testkit in two tests (discussed in https://github.com/akka/akka/pull/29241#issuecomment-644709436).

Also split the termination/folder-cleanup into separate methods.

ignasi35 committed:
Fix compile error

jrudolph commented:
> @jrudolph I've changed the test code to avoid using the same testkit in two tests (discussed in [#29241 (comment)](https://github.com/akka/akka/pull/29241#issuecomment-644709436)).
> 
> Also split the termination/folder-cleanup into separate methods.

Sounds good, I don't know much about the test, so I'm fine with any kind potential improvement you see fit.

jrudolph commented on akka-remote/src/test/scala/akka/remote/artery/tcp/ssl/RotatingKeysSSLEngineProviderSpec.scala:
```diff
@@ -261,11 +266,18 @@ abstract class RotatingKeysSSLEngineProviderSpec(extraConfig: String)
+    super.beforeTermination()
+  }
+
+  override def afterTermination(): Unit = {
```
Is it guaranteed that `afterTermination` runs when `beforeTermination` fails?

raboof committed:
Make naming easier to follow, avoid testkit sys more
